pkg/scheduler/partition.go

```diff
@@ -159,17 +159,20 @@ func (pc *PartitionContext) updatePreemption(conf configs.PartitionConfig) {
 }
 
 func (pc *PartitionContext) updatePartitionDetails(conf configs.PartitionConfig) error {
-	pc.Lock()
-	defer pc.Unlock()
+	// the following piece of code (before pc.Lock()) must be performed without locking
+	// to avoid lock order differences between PartitionContext and AppPlacementManager
 	if len(conf.Queues) == 0 || conf.Queues[0].Name != configs.RootQueue {
 		return fmt.Errorf("partition cannot be created without root queue")
 	}
 	log.Log(log.SchedPartition).Info("Updating placement manager rules on config reload")
-	err := pc.placementManager.UpdateRules(conf.PlacementRules)
+	err := pc.getPlacementManager().UpdateRules(conf.PlacementRules)
 	if err != nil {
 		log.Log(log.SchedPartition).Info("New placement rules not activated, config reload failed", zap.Error(err))
 		return err
 	}
+
+	pc.Lock()
+	defer pc.Unlock()
 	pc.rules = &conf.PlacementRules
```

Comment:
We should get rid of this. It does not serve any use anymore. When we moved to use placement rules always this field became obsolete. The content is also out of sync with the real set of rules in the placement manager.
Filed a follow up to cleanup and expose the placement rules in the statedump.

Reply:
Filed https://issues.apache.org/jira/browse/YUNIKORN-2554